### ISSUE TITLE
`cf push` app status info stack lists "unknown"

### ISSUE THREAD

**Issue by dmikusa-pivotal** [cloudfoundry/cli#799]
### Command

When I push an app to CF I get this output at the bottom after the app starts.

```
requested state: started
instances: 1/1
usage: 256M x 1 instances
urls: sinatra-test-dan.cfapps.io
last uploaded: Wed Mar 16 14:54:50 UTC 2016
stack: unknown
buildpack: https://github.com/cloudfoundry/ruby-buildpack#v1.6.5

     state     since                    cpu    memory        disk          details
#0   running   2016-03-16 10:55:12 AM   0.1%   21M of 256M   68.3M of 1G
```

The stack always lists "unknown.  This only happens with `cf push`.  The output of `cf app` shows the stack just fine.

### CLI Version

  - cf version 6.16.1+924508c-2016-02-26
  - cf v231
  - diego 0.1454.0
  - garden 0.333.0

I didn't test, but it's reported that this worked consistently on v6.14.

http://cf-dev.70369.x6.nabble.com/cf-dev-cf-push-stack-unknown-tp4206p4208.html

### CF Trace

```
REQUEST: [2016-03-17T22:43:00-04:00]
GET /v2/apps/11ff47a3-8e32-4c74-a7b0-85b61b87e00b/summary HTTP/1.1
Host: api.run.pivotal.io
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.16.1+924508c / darwin



RESPONSE: [2016-03-17T22:43:00-04:00]
HTTP/1.1 200 OK
Content-Length: 2318
Connection: keep-alive
Content-Type: application/json;charset=utf-8
Date: Fri, 18 Mar 2016 02:43:00 GMT
Server: nginx
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: e16da7d1-398d-4cd4-5723-a7bf80ccaa75
X-Vcap-Request-Id: 02faa379-502c-4dc2-5a00-0634efd491f8::d6c4fd4e-5db0-4428-a1e7-60f5dcdd8ac4

{"guid":"11ff47a3-8e32-4c74-a7b0-85b61b87e00b","name":"spring-music","routes":[{"guid":"f461c41a-2f57-439c-9b10-a142ec55cde1","host":"spring-music-dan","path":"","domain":{"guid":"fb6bd89f-2ed9-49d4-9ad1-97951a573135","name":"cfapps.io"}}],"running_instances":1,"services":[],"available_domains":[{"guid":"d576a360-a8f8-4aba-b15e-5b4787183784","name":"running-on-pws.com","owning_organization_guid":"639f74aa-5d97-4a47-a6b3-e9c2613729d8"},{"guid":"17d8c2dd-ee98-4ebd-b511-34c4b93d4ec0","name":"xxxxxx.com","owning_organization_guid":"639f74aa-5d97-4a47-a6b3-e9c2613729d8"},{"guid":"fb6bd89f-2ed9-49d4-9ad1-97951a573135","name":"cfapps.io","router_group_guid":null,"router_group_types":null}],"production":false,"space_guid":"10180e2b-33b9-44ee-9f8f-da96da17ac1a","stack_guid":"86205f38-84fc-4bc2-b2b8-af7f55669f04","buildpack":null,"detected_buildpack":"java-buildpack=v3.6-https://github.com/cloudfoundry/java-buildpack.git#5194155 open-jdk-like-jre=1.8.0_73 open-jdk-like-memory-calculator=2.0.1_RELEASE spring-auto-reconfiguration=1.10.0_RELEASE tomcat-access-logging-support=2.5.0_RELEASE tomcat-instan...","environment_json":{"JBP_CONFIG_OPEN_JDK_JRE":"[memory_calculator: {memory_sizes: {stack: 228k}}]"},"memory":512,"instances":1,"disk_quota":1024,"state":"STARTED","version":"d8057180-78ee-419c-85fb-a6876009267e","command":null,"console":false,"debug":null,"staging_task_id":"9734962bbd7e412d8f31e3d89e66d3db","package_state":"STAGED","health_check_type":"port","health_check_timeout":null,"staging_failed_reason":null,"staging_failed_description":null,"diego":true,"docker_image":null,"package_updated_at":"2016-03-18T02:42:15Z","detected_start_command":"CALCULATED_MEMORY=$($PWD/.java-buildpack/open_jdk_jre/bin/java-buildpack-memory-calculator-2.0.1_RELEASE -memorySizes=metaspace:64m..,stack:228k -memoryWeights=heap:75,metaspace:10,native:10,stack:5 -memoryInitials=heap:100%,metaspace:100% -totMemory=$MEMORY_LIMIT) &&  JAVA_HOME=$PWD/.java-buildpack/open_jdk_jre JAVA_OPTS=\"-Djava.io.tmpdir=$TMPDIR -XX:OnOutOfMemoryError=$PWD/.java-buildpack/open_jdk_jre/bin/killjava.sh $CALCULATED_MEMORY -Daccess.logging.enabled=false -Dhttp.port=$PORT\" exec $PWD/.java-buildpack/tomcat/bin/catalina.sh run","enable_ssh":true,"docker_credentials_json":{"redacted_message":"[PRIVATE DATA HIDDEN]"},"ports":[8080]}

REQUEST: [2016-03-17T22:43:00-04:00]
GET /v2/apps/11ff47a3-8e32-4c74-a7b0-85b61b87e00b/instances HTTP/1.1
Host: api.run.pivotal.io
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.16.1+924508c / darwin



RESPONSE: [2016-03-17T22:43:00-04:00]
HTTP/1.1 200 OK
Content-Length: 55
Connection: keep-alive
Content-Type: application/json;charset=utf-8
Date: Fri, 18 Mar 2016 02:43:00 GMT
Server: nginx
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: edf7e334-1cf9-4265-7c96-84ab87bbd845
X-Vcap-Request-Id: 78f9f0a2-7096-42fe-5f9e-7fa308e7f102::db55594b-5126-4434-a3a6-de9cfe2b0dc1

{"0":{"state":"RUNNING","uptime":1,"since":1458268979}}

REQUEST: [2016-03-17T22:43:00-04:00]
GET /v2/apps/11ff47a3-8e32-4c74-a7b0-85b61b87e00b/stats HTTP/1.1
Host: api.run.pivotal.io
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.16.1+924508c / darwin



RESPONSE: [2016-03-17T22:43:00-04:00]
HTTP/1.1 200 OK
Content-Length: 273
Connection: keep-alive
Content-Type: application/json;charset=utf-8
Date: Fri, 18 Mar 2016 02:43:00 GMT
Server: nginx
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 2c9a2591-8b83-485e-4967-73fbb181826f
X-Vcap-Request-Id: 2b97602f-4582-4b55-5b71-8dd20f3f3efe::514a4e89-477e-48b5-b98b-3ad6efbb66df

{"0":{"state":"RUNNING","stats":{"name":"spring-music","uris":["spring-music-dan.cfapps.io"],"host":"10.10.115.81","port":64283,"uptime":1,"mem_quota":536870912,"disk_quota":1073741824,"fds_quota":16384,"usage":{"time":"2016-03-18 02:43:00 UTC","cpu":0,"mem":0,"disk":0}}}}
OK

requested state: started
instances: 1/1
usage: 512M x 1 instances
urls: spring-music-dan.cfapps.io
last uploaded: Fri Mar 18 02:42:15 UTC 2016
stack: unknown
buildpack: java-buildpack=v3.6-https://github.com/cloudfoundry/java-buildpack.git#5194155 open-jdk-like-jre=1.8.0_73 open-jdk-like-memory-calculator=2.0.1_RELEASE spring-auto-reconfiguration=1.10.0_RELEASE tomcat-access-logging-support=2.5.0_RELEASE tomcat-instan...

     state     since                    cpu    memory      disk      details
#0   running   2016-03-17 10:42:59 PM   0.0%   0 of 512M   0 of 1G
```

### Platform Details

Mac OS 10.11.3

### Shell

iTerm

**Comment by dkoper** [comment]
Thanks Dan, we'll investigate.

**Comment by dkoper** [comment]
This has now been fixed in cf CLI 6.21.0. Thanks again for taking the time to report it.
https://github.com/cloudfoundry/cli#downloads
